lib/sidekiq/api.rb

```diff
@@ -1105,13 +1105,17 @@ def each(&block)
       results = []
       Sidekiq.redis do |conn|
         procs = conn.sscan_each("processes").to_a
-        procs.sort.each do |key|
-          valid, workers = conn.pipelined { |pipeline|
-            pipeline.exists?(key)
+
+        all_workers = conn.pipelined do |pipeline|
+          procs.sort.each do |key|
```

Comment:
I think this will give the wrong results. You need to persist the sorted results or else the zip below will not interleave the results with the correct proc. In other words, just add `.sort` to the end of line 1107 and remove it here.

Reply:
Yes, sure 🤦 Thank you.